students/chieu_quach/lesson07/html_render.py

```diff
@@ -0,0 +1,353 @@
+                try:                
+                    content.render(out_file)
+                   
+                except AttributeError:
+                    if content == "DOCTYPE":
+                        pass
+                    elif self.tag == "meta":
+                        out_file.write(indentation(content, head_indent_txt))
+                    elif self.tag == "pz":
+                        out_file.write(indentation(stylename_p, head_indent_txt))
+                        out_file.write(indentation(content, p_indent_txt))
+                    elif self.tag == "ul":
+                        out_file.write(indentation(stylename_ul, head_indent_txt))
+                    elif self.tag == "li":
+                        out_file.write(indentation(content, indent_txt))
+                    elif self.tag == "liu":
+                        out_file.write(indentation(stylename_li, p_indent_txt))
+                        out_file.write(indentation(content, indent_txt))
+                    else:
+                        out_file.write(content)
+                
+                    # if tag == title write output in one line including <title>
+                    if self.tag != "title":
+                        out_file.write("\n")
+                    
+
+           #   print ("self.tag ", self.tag)      
+           # prints the output content message             
+           #   print ("self.contents ", self.contents)
+            if self.tag == "liu":
+                self.tag = "li"
+            if self.tag == "pz":
+                self.tag = "p"
+
+            nameformat = "</{}>\n".format(self.tag)
+            if self.tag == "hr":
+          
+                out_file.write(indentation(nameformat, head_indent_txt))  
+           # out_file.write("<{} />\n".format(self.tag))
+           #   out_file.write(indentation(nameformat, head_indent_txt))
+            elif self.tag == "meta":
+                pass
+            elif self.tag == "head":
+         #   nameformat = "</{}>\n".format(self.tag)
+                out_file.write(indentation(nameformat, head_indent))
+            elif self.tag == "p":
+        #    nameformat = "</{}>\n".format(self.tag)
+                out_file.write(indentation(nameformat, head_indent_txt))
+            elif self.tag == "li":
+                out_file.write(indentation(nameformat, p_indent_txt))
+            elif self.tag == "ul":
+                out_file.write(indentation(nameformat, head_indent_txt))
+            elif self.tag == "body":
+                out_file.write(indentation(nameformat, head_indent))
+            else:
+            
+                out_file.write("</{}>\n".format(self.tag))
+
+
+
+            
+#     #   out_file.write("just something as a place holder...")
+
+class Meta(Element):
+    tag = "meta"
+
+class Html(Element):
+
+    tag = "html"
+  #  tag = "!DOCTYPE html"
+          
+    
+class Body(Element):
+    tag = "body"
+    
+class A(object):
```

Comment:
This class should be subclassing and overriding the `__init__` method to set your link (via kwargs)

Reply:
shouldn't need any other methods